yt_dlp/extractor/videocampus_sachsen.py

```diff
@@ -157,30 +197,33 @@ def _real_extract(self, url):
-        if not video_id:
-            video_id = embed_id or self._html_search_regex(
-                rf'src="https?://{host}/media/embed.*(?:\?|&)key=([0-9a-f]+)&?',
-                webpage, 'video_id')
+        formats, subtitles = [], {}
+        title = description = thumbnail = None
+
+        metadata = self._search_json(r'var\s+options\s*=\s*', webpage, 'player options', video_id or tmp_id or embed_id, default=None, fatal=False)
+
+        if metadata:
+            for source in metadata.get('sources', []):
+                if source.get('type') == 'application/x-mpegURL' and source.get('src'):
+                    _formats, _subtitles = self._extract_m3u8_formats_and_subtitles(source.get('src'), video_id or tmp_id or embed_id, fatal=False)
+                    formats.extend(_formats)
+                    subtitles.update(_subtitles)
+                elif source.get('src'):
+                    formats.append({'url': source.get('src')})
 
-        if not (display_id or tmp_id):
-            # Title, description from embedded page's meta wouldn't be correct
-            title = self._html_search_regex(r'<video-js[^>]* data-piwik-title="([^"<]+)"', webpage, 'title', fatal=False)
-            description = None
-            thumbnail = None
+            thumbnail = f'https://{host}{metadata.get("poster")}'
+            video_id = traverse_obj(metadata, ('videojsVimpOptions', 'Mediakey'))
         else:
-            title = self._html_search_meta(('og:title', 'twitter:title', 'title'), webpage, fatal=False)
-            description = self._html_search_meta(
-                ('og:description', 'twitter:description', 'description'), webpage, fatal=False)
             thumbnail = self._html_search_meta(('og:image', 'twitter:image'), webpage, fatal=False)
+            description = self._html_search_meta(('og:description', 'twitter:description', 'description'), webpage, fatal=False)
+            title = self._html_search_meta(('og:title', 'twitter:title', 'title'), webpage, fatal=False)
 
-        formats, subtitles = [], {}
-        try:
-            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
-                f'https://{host}/media/hlsMedium/key/{video_id}/format/auto/ext/mp4/learning/0/path/m3u8',
-                video_id, 'mp4', m3u8_id='hls', fatal=True)
-        except ExtractorError as e:
-            if not isinstance(e.cause, HTTPError) or e.cause.status not in (404, 500):
-                raise
+        if not title:
+            title = self._html_search_regex(r'<h1[^>]*>([^<]+)</h1>', webpage, 'title', default=None, fatal=False)
+        if not title:
+            embed_data = self._download_json(f'https://{host}/media/embedCode', video_id, data=f'key={video_id}'.encode(), fatal=False)
+            if embed_data:
+                title = self._html_search_regex(r'title="([^"]+)"', embed_data.get('embedCode', ''), 'title', fatal=False)
 
         formats.append({'url': f'https://{host}/getMedium/{video_id}.mp4'})
```

Comment:
we may not have `video_id` at this point
```suggestion
        if video_id:
            formats.append({'url': f'https://{host}/getMedium/{video_id}.mp4'})
        else:
            self.raise_no_formats('Unable to extract video ID')
```